packages/@aws-cdk/aws-efs/lib/access-point.ts

```diff
@@ -20,11 +20,16 @@ export interface IAccessPoint extends IResource {
    * @attribute
    */
   readonly accessPointArn: string;
+
+  /**
+   * The efs filesystem
+   */
+  readonly fileSystem?: IFileSystem;
```

Comment:
How about make this mandatory and throw an error in the implementation of `fromAccessPointId()`? 

The error message can be "fileSystemId is not available when 'fromAccessPointId()' is used. Use 'fromAccessPointAttributes()' instead"

Reply:
@nija-at, that would also be a breaking change as well, right ?

I'm wondering if there is a use-case (outside of EFS for lambda) where we would need to import an AccessPoint without having the need for `FileSystem` ?

Reply:
efs is an experimental module. Breaking changes are acceptable as long as they are [called out in the commit description](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md#step-4-commit).

> I'm wondering if there is a use-case (outside of EFS for lambda) where we would need to import an AccessPoint without having the need for FileSystem ?

In such cases exist, the `fromAccessPointId()` API can be used.

Reply:
I see. I didn't get the idea of how to implement this at start. I think I'm good now. I'm doing the same as in the rds imported cluster and throw the exception on the Getter when fileSystem is not present.. See below. Will update the PR.

```ts
class ImportedAccessPoint extends AccessPointBase {
  public readonly accessPointId: string;
  public readonly accessPointArn: string;
  private readonly _fileSystem?: IFileSystem;

  constructor(scope: Construct, id: string, attrs: AccessPointAttributes) {
    super(scope, id);

    if (!attrs.accessPointId) {
      if (!attrs.accessPointArn) {
        throw new Error('One of accessPointId or AccessPointArn is required!');
      }

      this.accessPointArn = attrs.accessPointArn;
      let maybeApId = Stack.of(scope).parseArn(attrs.accessPointArn).resourceName;

      if (!maybeApId) {
        throw new Error('ARN for AccessPoint must provide the resource name.');
      }

      this.accessPointId = maybeApId;
    } else {
      if (attrs.accessPointArn) {
        throw new Error('Only one of accessPointId or AccessPointArn can be provided!');
      }

      this.accessPointId = attrs.accessPointId;
      this.accessPointArn = Stack.of(scope).formatArn({
        service: 'elasticfilesystem',
        resource: 'access-point',
        resourceName: attrs.accessPointId,
      });
    }

    this._fileSystem = attrs.fileSystem;
  }

  public get fileSystem() {
    if (!this._fileSystem) {
      throw new Error("fileSystem is not available when 'fromAccessPointId()' is used. Use 'fromAccessPointAttributes()' instead");
    }

    return this._fileSystem;
  }
}
```